tests/ltp/patches/ltp_truncate_truncate03_fix.patch

```diff
@@ -0,0 +1,49 @@
++ Modified the tests by changing the temporary directory creation/deletion with root user.
++ Issue 297: [Tests] lkl_access_ok() should return -1 on invalid access
++ https://github.com/lsds/sgx-lkl/issues/297
+diff --git a/testcases/kernel/syscalls/truncate/truncate03.c b/testcases/kernel/syscalls/truncate/truncate03.c
+index 401c16a8e..316fb882c 100644
+--- a/testcases/kernel/syscalls/truncate/truncate03.c
++++ b/testcases/kernel/syscalls/truncate/truncate03.c
+@@ -76,7 +76,7 @@ static struct test_case_t {
+ } test_cases[] = {
+        { TEST_FILE1, TRUNC_LEN, EACCES },
+        { TEST_FILE2, TRUNC_LEN, ENOTDIR },
+-       { NULL, TRUNC_LEN, EFAULT },
++//     { NULL, TRUNC_LEN, EFAULT }, TODO: Enable once git issue 297 is fixed
+        { long_pathname, TRUNC_LEN, ENAMETOOLONG },
+        { "", TRUNC_LEN, ENOENT },
+        { TEST_DIR1, TRUNC_LEN, EISDIR },
+@@ -121,14 +121,12 @@ void setup(void)
+        tst_sig(NOFORK, DEF_HANDLER, cleanup);
+
+        tst_require_root();
+-
++       tst_tmpdir();
```

Comment:
why does this need to be moved earlier in the execution?

Reply:
is this related to having it created by the root user?

Reply:
nobody user is failing to create directory under /tmp, with root we can create. As it does not effect test purpose so move it to earlier.